qiskit/transpiler/target.py

```diff
@@ -373,7 +381,15 @@ def add_instruction(self, instruction, properties=None, name=None):
         self._instruction_schedule_map = None
 
     def update_instruction_properties(self, instruction, qargs, properties):
-        """Update the property object for an instruction qarg pair already in the Target
+        """Update the property object for an instruction qarg pair already in the Target.
+
+        For ease of access, a user is able to obtain the mapping between an instruction's
+        applicable qargs and its instruction properties via indexing (using ``__getitem__``),
+        with the instruction's name as the key. However, due to the strictly additive nature
+        of the ``Target``'s mappings and their complexity, this method is the only way to
+        modify/update the properties of an instruction in the ``Target``. Usage of indexing
+        for modifications (via usage of ``__setitem__`` on this or any submaps of the ``Target``)
+        is not supported.
```

Comment:
```suggestion
        with the instruction's name as the key. This method is the only way to
        modify/update the properties of an instruction in the ``Target``. Usage of the mapping protocol
        for modifications is not supported.
```

The extra details don't need to be on this method, they should be in the top level class documentation